openstack/identity/identity.go

```diff
@@ -209,6 +212,7 @@ func (h Handler) checkToken(r *http.Request, tenant string, tokenGetResult getRe
 	/* TODO Caching or PKI */
 	for _, a := range h.ValidAdmins {
 		if validateProjectRole(tokenGetResult, a.Project, a.Role) == true {
+			context.Set(r, service.PrivKey, "true")
```

Comment:
I don't think we need to use gorilla context.  The standard library now has everything we need.  If I understand correctly, what we're trying to do is to associate some information with a request object and retrieve this info at a later state.

This could be done as follows

r = r.WithContext(context.WithValue(r.Context(), service.PrivKey, "true")) 

Note you'd need to return r from this function and from validateToken.

Then in ServeHTTP, you'd do a 

ok := r.Context().Value(service.PrivKey)